pkg/testutils/logger_test.go

```diff
@@ -45,3 +46,28 @@ func TestJSONLineError(t *testing.T) {
 	_, ok := log.JSONLine(0)["error"]
 	assert.True(t, ok, "must have 'error' key")
 }
+
+// NB. Run with -race to ensure no race condition
+func TestRaceCondition(t *testing.T) {
```

Comment:
does this test panic when run with `-race` switch and without the lock in the Buffer (i.e. before your fix)?